Slack Connector Doc Edit (#1239)
mulesoft/docs-connectors: slack/1.0/modules/ROOT/Edit/slack-connector-studio.adoc

= Use Studio to Configure Slack Connector l.0 - Mule 4

Anypoint Studio (Studio) editors help you design and update your Mule applications, properties, and configuration files.

To add and configure a connector in Studio:

. <<create-mule-project,Create a Mule project>>.
. <<add-connector-to-project,Add the connector to your Mule project>>.
. <<configure-input-source,Configure a source for the connector's flow>>.
. <<add-connector-operation,Add a connector operation to the flow>>.
. <<configure-global-element,Configure a global element for the connector>>.

> . <<configure-other-fields,Configure the Other Connector Fields>>.

Reviewer: ```suggestion
. <<configure-other-fields,Configure the other connector fields>>.
```

When you run the connector, you can view the app log to check for problems in real time, as described in <<view-app-log,View the App Log>>.

If you are new to configuring connectors in Studio, see xref:connectors::introduction/intro-config-use-studio.adoc[Use Studio to Configure a Connector]. If, after reading this topic, you need additional information about the connector fields, see the xref:slack-connector-reference.adoc[Slack Connector Reference].

[[create-mule-project]]
== Create a Mule Project

In Studio, create a new Mule project in which to add and configure the connector:

. In Studio, select *File > New > Mule Project*.
. Enter a name for your Mule project and click *Finish*.

[[add-connector-to-project]]
== Add the Connector to Your Mule Project

Add Anypoint Connector for Slack (Slack Connector) to your Mule project to automatically populate the XML code with the connector's namespace and schema location and to add the required dependencies to the project's `pom.xml` file:

. In the *Mule Palette* view, click *(X) Search in Exchange*.

> . In the *Add Dependencies to Project* window, type `slack`` in the search field.

Reviewer: ```suggestion
. In the *Add Dependencies to Project* window, type `slack` in the search field.
```

. Click *Slack Connector* in *Available modules*.
. Click *Add*.
. Click *Finish*.

Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.

[[configure-input-source]]
== Configure a Source

A source initiates a flow when a specified condition is met.

> You can configure one of these sources to use with Slack Connector:

Reviewer: ```suggestion
Configure one of these sources to use with Slack Connector:
```

* HTTP Listener +
Initiates a flow each time it receives a request on the configured host and port
* Scheduler +

> Initiates a flow when a time-based condition is met

Reviewer: Add line-spaces as done in the list in Lines 192-210 above.

For example, to configure an HTTP Listener source, follow these steps:

. In the *Mule Palette* view, select *HTTP > Listener*.
. Drag *Listener* to the Studio canvas.
. On the *Listener* configuration screen, optionally change the value of the *Display Name* field.
. Specify a value for the *Path* field.
. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of the HTTP Listener source in the app.
. On the *General* tab, specify the connection information for the connector.
. On the *TLS* tab, optionally specify the TLS information for the connector.

> . On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.

Reviewer: I would use "in" here as well.

I'm going to see if I can find a table of prepositions to use with UI elements somewhere, to give us guidance.

When you configure a connector, it’s best to configure a global element that all instances of that connector in the app can use. Configuring a global element requires you to provide the authentication credentials that the connector requires to access the target Slack system. Slack Connector supports OAuth 2.0 authentication, which delegates user authentication to the service hosting the user account.

To configure the global element for Slack Connector, follow these steps:

. Select the name of the connector in the Studio canvas.
. Select the operation in the Studio canvas.
. In the configuration screen for the operation, click the plus sign (*+*) next to the *Connector configuration* field to access the global element configuration fields.
. On the *General* tab, complete the following fields to set up OAuth 2.0 authentication:
+
[%header%autowidth.spread]
|===
|Field |Description
|*Consumer key*| OAuth consumer key as registered with the service provider
|*Consumer secret*| OAuth consumer secret as registered with the service provider
|*Authorization url*| Service provider's authorization endpoint URL
|*Access token url*| Service provider's access token endpoint URL

> |*Scopes*| OAuth scopes to request during the OAuth dance. If not provided, this value defaults to those in the annotation.

Reviewer: Because this item contains a complete sentence, either:

- End each of the other Description column entries with a period.

- Reformat to look like this:

  OAuth scopes to request during the OAuth dance
  
  If not provided, this value defaults to those in the annotation.

|*Resource owner Id*| Resource owner ID that each component should use if it doesn't otherwise reference one
|*Before*| Name of a flow to execute immediately before starting the OAuth dance
|*After*| Name of a flow to execute immediately after an access token is received
|*Listener config*| Reference to an HTTP listener configuration to use to create the listener that receives the access token callback endpoint
|*Callback path* | Path of the access token callback endpoint
|*Authorize path* | Path of the local HTTP endpoint that triggers the OAuth dance

> |*External callback url* |If the callback endpoint is behind a proxy or must be accessed through a non-direct URL, use this parameter to tell the OAuth provider the URL to use to access the callback

Reviewer: parallelism
```suggestion
|*External callback url* |If the callback endpoint is behind a proxy or must be accessed through a indirect URL, a parameter that tells the OAuth provider the URL to use to access the callback
```

> |*Object store* | Reference to the object store to use to store each resource owner ID's data. If not specified, Mule automatically provisions the default object store.

Reviewer: ditto comment for *Scopes* definition above

|===
+
You can reference a configuration file that contains ANT-style property placeholders (recommended), or you can enter your authorization credentials in the global configuration properties. For information about the benefits of using property placeholders and how to configure them, see xref:connectors::introduction/intro-connector-configuration-overview.adoc[Anypoint Connector Configuration].
+
. On the *Advanced* tab, optionally specify reconnection information, including a reconnection strategy.
. Click *Test Connection* to confirm that Mule can connect with the specified server.
. Click *OK*.

[[configure-other-fields]]
== Configure Additional Connector Fields

> After you configure a global element for Slack Connector, configure the the other required fields for the connector.The required fields vary depending on which connector operation you use.

Reviewer: ```suggestion
After you configure a global element for Slack Connector, configure the the other required fields for the connector. The required fields vary depending on which connector operation you use.
```
